app/models/vehicles.py

```diff
@@ -54,16 +48,21 @@ def insert(
     # TODO: Upload images into disk
     pass
 
-    return vehicle_id
+    return id
 
 
 def delete(vehicle_id):
     query = tinydb.Query()
-    vehicles.remove(query.id == vehicle_id)
+    vehicles.remove(query.vehicle_id == vehicle_id)
 
 
 def all():
-    return vehicles.all()
+    results = vehicles.all()
+
+    if not results:
+        raise ErrVehicleNotFound("oops.. no vehicles found in the db")
```

Comment:
after thinking about it, I think this function should not raise an error in case no data was returned from the db